aptos-move/framework/src/natives/object.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use move_binary_format::errors::{PartialVMError, PartialVMResult};
+use move_core_types::{
+    account_address::AccountAddress, gas_algebra::InternalGas, vm_status::StatusCode,
+};
+use move_vm_runtime::native_functions::{NativeContext, NativeFunction};
+use move_vm_types::{
+    loaded_data::runtime_types::Type, natives::function::NativeResult, pop_arg, values::Value,
+};
+use smallvec::smallvec;
+use std::{collections::VecDeque, sync::Arc};
+
+/***************************************************************************************************
+ * native exists_at<T>
+ *
+ *   gas cost: base_cost
+ *
+ **************************************************************************************************/
+#[derive(Clone, Debug)]
+pub struct ExistsAtGasParameters {
+    pub base_cost: InternalGas,
```

Comment:
Add a TODO and open a issue to align the gas with that of the actual `exists`? Otherwise objects have some weird backdoor to an exists check with different gas behavior.